src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -3489,6 +3622,33 @@ emit_sri_vector_t (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *f
 		g_assert (sizeof (buf) >= size);
 		memset (buf, 0, sizeof (buf));
 
+#ifdef TARGET_S390X
+    /* we directly emit vrepi*/
+   if (etype->type != MONO_TYPE_R4 && etype->type != MONO_TYPE_R8)
+        return emit_simd_ins_for_sig (cfg, klass, OP_VEC_ONE, etype->type, 1, fsig, args);
```

Comment:
This is platform-specific code, why not just emit the proper vrepi instead of this pseudo OP_VEC_ONE ?